### PR TITLE
#1987 Enable option to allow link lines to be drawn over nodes using `enableLinksOverNodes` config.

### PR DESCRIPTION
**Pull request by srikant-ch5**
Fixes: #1987 

`enableLinksOverNodes` config can be changed using toggle in side panel.
Screenshot 2024-05-28 at 4 39 10 PM

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by srikant-ch5** [comment]
Hi @tomlyn Could you please review these changes. As per these changes link over nodes are appearing like below.
Screenshot 2024-05-30 at 4 41 22 PM

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js [hunk 1]
```diff
@@ -1976,7 +1976,9 @@ export default class SVGCanvasRenderer {
 					return;
 				}
 				const nodeGrp = d3.select(d3Event.currentTarget);
-				this.raiseNodeToTop(nodeGrp);
+				if (!this.config.enableLinksOverNodes) {
```
Comment (tomlyn) [inline canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js:1979]: It's not a big thing but, can you move this to the list of 'if' conditions inside the raiseNodeToTop function to keep all the conditions together.

**Review by tomlyn — CHANGES_REQUESTED** [review]
Looks good. I added one minor comment. Can you also add a section to https://elyra-ai.github.io/canvas/03.02.01-canvas-config/#links to document this new feature. Also, you'll need to update the release notes to mention this.

**Comment by tomlyn** [comment]
One further comment -- see my comment in the issue about changing the 'React Nodes - Mapping' sample app.

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js (hunk 1 above)
Comment (srikant-ch5) [reply canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js]: Sure, I have updated accordingly.

**Comment by srikant-ch5** [comment]
> One further comment -- see my comment in the issue about changing the 'React Nodes - Mapping' sample app.

Hi @tomlyn I have included above to change to let links pass over nodes in `React Nodes - Mapping` sample app. Please review.

**Review by tomlyn — CHANGES_REQUESTED** [review]

**Review by tomlyn — APPROVED** [review]
LGTM

**Pull request merged**